### ISSUE TITLE
postcss-simple-vars unknown variable when using nested loops

### ISSUE THREAD

**Issue by hendriku** [outpunk/postcss-each#41]
Hello there, we'd like to use this plugin together with postcss-simple-vars. This is supported, right?

I am currently getting this error in one of my postcss files:

![image](https://user-images.githubusercontent.com/8998518/137635366-071662ac-a48a-42cd-9432-5faa91646b1c.png)

with the following as my config:
```

module.exports = (ctx) => ({
  map: ctx.options.map,
  parser: ctx.options.parser,
  plugins: [
    require("postcss-import")(require("postcss-normalize")().postcssImport()),
    require("postcss-preset-env"),
    require("postcss-flexbugs-fixes"),
    require("postcss-mixins"),
    require("postcss-nested"),
    require("postcss-each")({
      plugins: {
        afterEach: [require("postcss-simple-vars"), require("postcss-calc")]
      }
    }),
    require("postcss-hexrgba"),
    require("postcss-utilities"),
    require("postcss-viewport-height-correction"),
    require("postcss-size"),
    require("autoprefixer"),
    ctx.env === "production" ? require("cssnano") : false
  ]
});
```

If I stop using postcss-simple-vars the postcss-each package works and I am getting no errors. But configured like this, postcss-simple-vars should actually not see the \@\each rules statements, shouldn't they?

Thanks in advance!

**Comment by hendriku** [comment]
Okay now I've managed to get a minimal example working and figured out some detail about the issue. **First of all I mixed up the plugin order.** afterEach is obviously called BEFORE the at-each statements are resolved. Maybe a note in the docs explaining this also on the example of simple-vars would be helpful to others. However, as simple-vars thinks the $i were its variables it starts to complain. 
My minimal example working with `beforeEach` instead of `afterEach` in postcss config:

```css
$black: #010101;
$white: #fff;

body {
  background-color: $black
}

@each $i in 1,2,4,8,16 {
  body.m$(i) {
    background-color: $white;
    margin: $(i)px;
  }
}
```

BUT there is another issue. As soon as I nest multiple each definitions, the error comes back within the inner at-each. This makes sense as after the resolving of the outer at-each simple-vars is being called and faces the same issue as mentioned initially. Example:


```css
$black: #010101;
$white: #fff;

body {
  background-color: $black
}

@each $i in 1,2,4,8,16 {
  @each $j in 1,2,3,4 {
    body.m$(i)-$(j) {
      background-color: $white;
      margin: $(i)px;
    }
  }
}
```

produces 

![image](https://user-images.githubusercontent.com/8998518/137676078-7bee30ce-39c2-4ea8-a395-f312c9be7529.png)


TLDR; How can we use this package together with @postcss-simple-vars AND use nested at-each statements? :) @madyankin

**Comment by ryansch** [comment]
After digging in to this issue I decided to configure simple-vars to emit a warning _only_ if the node exists after all of the plugins run.

```javascript
require('postcss-simple-vars')({
  unknown(node, name, result) {
    // Print out warning if the node still exists at the end.
    node.warn(result, 'Unknown variable ' + name);
  }
}),
```
